wikimedia/wikipedia-ios
# Add interruption levels and relevance scores to incoming notifications

**Phabricator:** https://phabricator.wikimedia.org/T285343

### Notes
Per the Echo Notification mapping spreadsheet in the description of the linked Phab ticket, adds passive, active, and time sensitive notification priorities to our supported types.

### Test Steps
You can manually assign these interruption levels and relevance scores to the `bestAttemptContent` in `NotificationService.swift` and use our Push utility to test these. Otherwise you can build to device, trigger the notifications types below, then use our Push utility to cause the push notification to fire on device:

1. Confirm you are logged into an account in the app with Settings > Push notifications enabled.
2. Tap your device's sleep/wake button to put it in sleep mode with the display off. Find an `active` notification type in our mapping doc, and trigger that notification type for the account in step 1. Confirm your device screen lights up with that notification displayed on the lock screen.
3. Tap your device's sleep/wake button to put it in sleep mode with the display off. Find a `passive` notification and trigger it on the same account. Confirm that your device screen _does not_ light up, but when you wake your device and visit iOS's notification center that this notification is visible.
4. Tap your device's sleep/wake button to put it in sleep mode with the display off. Find a `time sensitive` notification and trigger it on the same account. Confirm your device lights up displaying that notification, and that `Time Sensitive` is displayed in a label above the notification's title.
5. Enable a [Focus mode](https://support.apple.com/en-us/HT212608) and allow `Time Sensitive` notifications for that focus mode. Repeat Step 4 and confirm the time sensitive notification "breaks through" your focus mode and is displayed on your lock screen. Repeat Steps 2 and 3 and confirm the active and passive notifications _do not_ break through your focus mode and aren't displayed on your lock screen but are displayed in iOS's notification center.
6. Using that same Focus mode, disallow `Time Sensitive` notifications. Repeat Steps 2, 3, and 4 and confirm none of the notifications "break through" your focus mode on the lock screen. Confirm they are visible when your navigate to iOS's notification center.

![time_sensitive](https://user-images.githubusercontent.com/5652327/163877289-a637dd80-7f46-44de-88d4-02311996e573.PNG)
